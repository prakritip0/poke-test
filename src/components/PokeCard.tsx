import React from 'react';
import { Link } from 'react-router-dom';

interface PokeCardProps {
  name: string;
  routeName:string
}

const PokeCard = ({name, routeName}: PokeCardProps) => {
  console.log('Name', name, 'route', routeName);
  
  return (
    <Link data-testid='pokemon-card' to={`/details/${routeName}`}>
      <div className='bg-slate-100 px-3 py-5 rounded'>
        <p>{name}</p>
      </div>
    </Link>
  );
};

export default PokeCard;
