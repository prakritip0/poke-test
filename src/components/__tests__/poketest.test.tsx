import {findByText, render, screen} from '@testing-library/react';
import PokeCard from '../PokeCard';
import React from 'react';
import '@testing-library/jest-dom';

describe('Pokedex home', () => {
  const TestComponent = <PokeCard name='Bulbasaur' routeName='/details/bulbasur' />;
  it('Renders properly', () => {
    render(TestComponent);
    expect(screen.getByText('Bulbasaur')).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const {baseElement} = render(TestComponent);
    expect(baseElement).toMatchSnapshot();
  });
});
