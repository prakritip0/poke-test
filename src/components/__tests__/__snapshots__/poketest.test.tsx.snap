// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokedex home matches snapshot 1`] = `
<body>
  <div>
    <div
      class="bg-slate-100 px-3 py-5 rounded"
    >
      <p>
        lalala
      </p>
    </div>
  </div>
</body>
`;
